cmplxs/cmplxs.go

```diff
@@ -0,0 +1,679 @@
+// Copyright ©2013 The Gonum Authors. All rights reserved.
+// Use of this code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmplxs
+
+import (
+	"errors"
+	"math"
+	"math/cmplx"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/internal/asm/c128"
+)
+
+// Abs calculates the absolute values of the elements of s, and stores them in dst.
+// It panics if the lengths of dst and s do not match.
+func Abs(dst []float64, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of the slices do not match")
+	}
+	for i, v := range s {
+		dst[i] = cmplx.Abs(v)
+	}
+}
+
+// Add adds, element-wise, the elements of s and dst, and stores the result in dst.
+// It panics if the lengths of dst and s do not match.
+func Add(dst, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of the slices do not match")
+	}
+	c128.AxpyUnitaryTo(dst, 1, s, dst)
+}
+
+// AddTo adds, element-wise, the elements of s and t and
+// stores the result in dst. It panics if the lengths of s, t and dst do not match.
+func AddTo(dst, s, t []complex128) []complex128 {
+	if len(s) != len(t) {
+		panic("cmplxs: length of adders do not match")
+	}
+	if len(dst) != len(s) {
+		panic("cmplxs: length of destination does not match length of adder")
+	}
+	c128.AxpyUnitaryTo(dst, 1, s, t)
+	return dst
+}
+
+// AddConst adds the scalar c to all of the values in dst.
+func AddConst(c complex128, dst []complex128) {
+	c128.AddConst(c, dst)
+}
+
+// AddScaled performs dst = dst + alpha * s.
+// It panics if the lengths of dst and s are not equal.
+func AddScaled(dst []complex128, alpha complex128, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of destination and source to not match")
+	}
+	c128.AxpyUnitaryTo(dst, alpha, s, dst)
+}
+
+// AddScaledTo performs dst = y + alpha * s, where alpha is a scalar,
+// and dst, y and s are all slices.
+// It panics if the lengths of dst, y, and s are not equal.
+//
+// At the return of the function, dst[i] = y[i] + alpha * s[i]
+func AddScaledTo(dst, y []complex128, alpha complex128, s []complex128) []complex128 {
+	if len(dst) != len(s) || len(dst) != len(y) {
+		panic("cmplxs: lengths of slices do not match")
+	}
+	c128.AxpyUnitaryTo(dst, alpha, s, y)
+	return dst
+}
+
+// Count applies the function f to every element of s and returns the number
+// of times the function returned true.
+func Count(f func(complex128) bool, s []complex128) int {
+	var n int
+	for _, val := range s {
+		if f(val) {
+			n++
+		}
+	}
+	return n
+}
+
+// CumProd finds the cumulative product of the first i elements in
```

Comment:
The wording here suggests that there is a parameter `i` that specifies how many elements of `s` are used. I would change it to something like
```
// CumProd finds the cumulative product of elements of s and
// stores it in place into dst so that
//  dst[i] = s[i] * s[i-1] * s[i-2] * ... * s[0]
// ...
```
The same for `CumSum`.

Reply:
Done

Will also need to be done in floats.